godot-core/src/builtin/aabb.rs

```diff
@@ -43,32 +43,346 @@ impl Aabb {
+    pub fn is_finite(&self) -> bool {
+        self.position.is_finite() && self.size.is_finite()
+    }
+
+    /// The end of the `Aabb` calculated as `position + size`.
     #[inline]
     pub fn end(&self) -> Vector3 {
         self.position + self.size
     }
 
     /// Set size based on desired end-point.
-    ///
-    /// _Godot equivalent: `Aabb.size` property_
     #[inline]
     pub fn set_end(&mut self, end: Vector3) {
         self.size = end - self.position
     }
 
     /// Returns `true` if the two `Aabb`s are approximately equal, by calling `is_equal_approx` on
     /// `position` and `size`.
-    ///
-    /// _Godot equivalent: `Aabb.is_equal_approx()`_
     #[inline]
     pub fn is_equal_approx(&self, other: &Self) -> bool {
         self.position.is_equal_approx(other.position) && self.size.is_equal_approx(other.size)
     }
 
-    /* Add in when `Aabb::abs()` is implemented.
+    /// Returns the normalized longest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis()`_
+    pub fn longest_axis(&self) -> Vector3 {
+        match self.longest_axis_index() {
+            Vector3Axis::X => Vector3::RIGHT,
+            Vector3Axis::Y => Vector3::UP,
+            Vector3Axis::Z => Vector3::BACK,
+        }
+    }
+
+    /// Returns the index of the longest axis of the AABB (according to Vector3's AXIS_* constants).
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis_index()`_
+    pub fn longest_axis_index(&self) -> Vector3Axis {
+        self.size.max_axis_index()
+    }
+
+    /// Returns the scalar length of the longest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_longest_axis_size()`_
+    pub fn longest_axis_size(&self) -> real {
+        self.size.x.max(self.size.y.max(self.size.z))
+    }
+
+    /// Returns the normalized shortest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis()`_
+    pub fn shortest_axis(&self) -> Vector3 {
+        match self.shortest_axis_index() {
+            Vector3Axis::X => Vector3::RIGHT,
+            Vector3Axis::Y => Vector3::UP,
+            Vector3Axis::Z => Vector3::BACK,
+        }
+    }
+
+    /// Returns the index of the shortest axis of the AABB (according to Vector3::AXIS* enum).
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis_index()`_
+    pub fn shortest_axis_index(&self) -> Vector3Axis {
+        self.size.min_axis_index()
+    }
+
+    /// Returns the scalar length of the shortest axis of the AABB.
+    ///
+    /// _Godot equivalent: `AABB.get_shortest_axis_size`_
+    pub fn shortest_axis_size(&self) -> real {
+        self.size.x.min(self.size.y.min(self.size.z))
+    }
+
+    /// Returns the support point in a given direction. This is useful for collision detection algorithms.
+    ///
+    /// _Godot equivalent: `AABB.get_support()`_
+    pub fn support(&self, dir: Vector3) -> Vector3 {
+        let half_extents = self.size * 0.5;
+        let ofs = self.position + half_extents;
```

Comment:
I think I just copied Godot's C++ on this one, I really have no idea 😭

Reply:
Well, it would be good to understand the code 😬

Some things gdext does differently than Godot's C++ impl, and since we don't use FFI calls but a rust-native implementation, we will also have to maintain this code.

Reply:
Well it seems like it's the center of the Aabb, I will change this.

Reply:
Since you added a `center()` function, let's just reuse that.